opentracing-util/src/main/java/io/opentracing/util/ThreadLocalScopeManager.java

```diff
@@ -19,11 +19,33 @@
 
 /**
  * A simple {@link ScopeManager} implementation built on top of Java's thread-local storage primitive.
+ * <p>
+ * Optionally supports {@link ScopeListener}, to perform additional actions when scope is changed for given thread.
+ * Listener methods are always called synchronously on the same thread, right after activation (meaning {@link #active()}
+ * already returns new a scope).
  *
  * @see ThreadLocalScope
  */
 public class ThreadLocalScopeManager implements ScopeManager {
+
     final ThreadLocal<ThreadLocalScope> tlsScope = new ThreadLocal<ThreadLocalScope>();
+    final ScopeListener listener;
```

Comment:
Consider making it a list or adding a `CompoundScopeListener`, which maintains a list of `ScopeListeners`

Reply:
composite listener can be another PR, imo, if someone actually needs it.